src/components/CommentCard.jsx

```diff
@@ -1,14 +1,50 @@
 import { dateConverter } from "../utils/utils";
+import { useContext, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteCommentByCommentId } from "../api";
 
 export default function CommentCard(props) {
-  const { comment } = props;
+  const { comment, setComments } = props;
+  const { user } = useContext(UserContext);
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleDeleteComment(event) {
+    event.target.disabled = true;
+    setIsLoading(() => {
+      return true;
+    });
+    deleteCommentByCommentId(comment.comment_id).then(() => {
+      setComments((currentComments) => {
+        const indexToSplice = currentComments.findIndex((currComment) => {
+          return currComment.comment_id === comment.comment_id;
+        });
+        const splicedComments = currentComments.toSpliced(indexToSplice, 1);
+
+        return [...splicedComments];
+      });
+      setIsLoading(() => {
+        return false;
+      });
+    });
+  }
 
   return (
     <article className="comment-card">
       <p className="comment-author">{comment.author}</p>
-      <p className="comment-body">{comment.body}</p>
+      <section className="comment-body">
+        {comment.body.split("\n").map((commentSegment, index) => {
```

Comment:
Why is this necessary?

Reply:
Hi Mark, any enters in a posted comments text area were being joint together so this was to allow the comments to be formatted in paragraph styles